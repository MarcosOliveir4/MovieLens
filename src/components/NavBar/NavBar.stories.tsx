import { Meta, StoryObj } from '@storybook/react'
import NavBar from '.'

export default {
  title: 'NavBar',
  component: NavBar
} as Meta

export const Default: StoryObj = {}
