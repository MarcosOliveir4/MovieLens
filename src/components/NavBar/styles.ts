'use client'
import styled from 'styled-components'

export const Nav = styled.nav`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: ${({ theme }) => theme.colors.softBlack};
`

export const H2 = styled.h2``

export const P = styled.p`
  gap: 0.5rem;
  display: flex;
  cursor: pointer;

  text-decoration: none;
  color: ${({ theme }) => theme.colors.yellow};
  transition: color 0.5s;

  &:hover {
    color: ${({ theme }) => theme.colors.darkYellow};
  }
`

export const Form = styled.form`
  display: flex;
  gap: 0.5rem;
`
export const Input = styled.input`
  padding: 0.2rem 0.8rem;
  border-radius: 4px;
  border: none;

  &:focus {
    outline: 2px solid ${({ theme }) => theme.colors.yellow};
  }
`
export const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.yellow};
  border: 2px solid ${({ theme }) => theme.colors.yellow};
  border-radius: 4px;
  color: ${({ theme }) => theme.colors.black};
  padding: 0.3rem;
  font-size: 1.3rem;

  display: flex;
  align-items: center;
  cursor: pointer;

  transition: all 0.5s;
  &:hover {
    color: ${({ theme }) => theme.colors.yellow};
    background-color: transparent;
  }
`
