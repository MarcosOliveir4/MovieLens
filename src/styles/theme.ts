const Theme = {
  background: '#000',
  colors: {
    warning: '#f7d354',
    darkWarning: '#b8930c',

    fontColor: '#fff'
  }
}

export default Theme
