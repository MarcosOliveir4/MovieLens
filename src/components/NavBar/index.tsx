'use client'

import * as S from './styles'

import { BiCameraMovie, BiSearchAlt2 } from 'react-icons/bi'
import React, { useState } from 'react'
import { useRouter } from 'next/navigation'

const NavBar = () => {
  const router = useRouter()
  const [search, setSearch] = useState<string>('')

  const handleSearch = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()
    router.push(`/home?q=${search}`)
    setSearch('')
  }

  return (
    <S.Nav data-testid="navbar">
      <S.H2>
        <S.P
          onClick={() => router.push('/home')}
          data-testid="navbar-link-home"
        >
          <BiCameraMovie /> {'MovieLens'}
        </S.P>
      </S.H2>

      <S.Form onSubmit={handleSearch} data-testid="navbar-form">
        <S.Input
          data-testid="navbar-form-input"
          type="text"
          placeholder="Busque um filme"
          value={search}
          onChange={(event) => setSearch(event.target.value)}
        />
        <S.Button type="submit" data-testid="navbar-form-button">
          <BiSearchAlt2 />
        </S.Button>
      </S.Form>
    </S.Nav>
  )
}

export default NavBar
