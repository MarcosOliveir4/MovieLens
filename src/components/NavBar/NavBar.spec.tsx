import { render, screen } from '@testing-library/react'
import NavBar from '.'
import theme from '@/styles/theme'
import { ThemeProvider } from 'styled-components'
import userEvent from '@testing-library/user-event'

const mockPush = jest.fn()

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      prefetch: () => null,
      push: mockPush
    }
  }
}))

describe('<NavBar />', () => {
  beforeEach(() => {
    render(
      <ThemeProvider theme={theme}>
        <NavBar />
      </ThemeProvider>
    )
  })
  afterEach(() => {
    jest.clearAllMocks()
  })
  it('should render the heading', () => {
    const navbar = screen.getByTestId('navbar')
    expect(navbar).toMatchSnapshot()
  })

  it('should emit onSearch when form is submitted', async () => {
    const formInput = screen.getByTestId('navbar-form-input')
    const formButton = screen.getByTestId('navbar-form-button')

    await userEvent.type(formInput, 'Spirited Away')
    await userEvent.click(formButton)

    expect(mockPush).toHaveBeenCalledTimes(1)
    expect(mockPush).toHaveBeenCalledWith('/home?q=Spirited Away')
  })

  it('should redirect to the home when the logo is clicked on', async () => {
    const logo = screen.getByTestId('navbar-link-home')
    await userEvent.click(logo)
    expect(mockPush).toHaveBeenCalledTimes(1)
    expect(mockPush).toHaveBeenCalledWith('/home')
  })
})
