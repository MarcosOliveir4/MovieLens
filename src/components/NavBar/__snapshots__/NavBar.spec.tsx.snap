// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render the heading 1`] = `
.c0 {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: #121212;
}

.c1 {
  gap: 0.5rem;
  display: flex;
  cursor: pointer;
  text-decoration: none;
  color: #f7d354;
  transition: color 0.5s;
}

.c1:hover {
  color: #b8930c;
}

.c2 {
  display: flex;
  gap: 0.5rem;
}

.c3 {
  padding: 0.2rem 0.8rem;
  border-radius: 4px;
  border: none;
}

.c3:focus {
  outline: 2px solid #f7d354;
}

.c4 {
  background-color: #f7d354;
  border: 2px solid #f7d354;
  border-radius: 4px;
  color: #000;
  padding: 0.3rem;
  font-size: 1.3rem;
  display: flex;
  align-items: center;
  cursor: pointer;
  transition: all 0.5s;
}

.c4:hover {
  color: #f7d354;
  background-color: transparent;
}

<nav
  class="c0"
  data-testid="navbar"
>
  <h2
    class=""
  >
    <p
      class="c1"
      data-testid="navbar-link-home"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 11c0-.959-.68-1.761-1.581-1.954C16.779 8.445 17 7.75 17 7c0-2.206-1.794-4-4-4-1.517 0-2.821.857-3.5 2.104C8.821 3.857 7.517 3 6 3 3.794 3 2 4.794 2 7c0 .902.312 1.727.817 2.396A1.994 1.994 0 0 0 2 11v8c0 1.103.897 2 2 2h12c1.103 0 2-.897 2-2v-2.638l4 2v-7l-4 2V11zm-5-6c1.103 0 2 .897 2 2s-.897 2-2 2-2-.897-2-2 .897-2 2-2zM6 5c1.103 0 2 .897 2 2s-.897 2-2 2-2-.897-2-2 .897-2 2-2zM4 19v-8h12l.002 8H4z"
        />
      </svg>
       
      MovieLens
    </p>
  </h2>
  <form
    class="c2"
    data-testid="navbar-form"
  >
    <input
      class="c3"
      data-testid="navbar-form-input"
      placeholder="Busque um filme"
      type="text"
      value=""
    />
    <button
      class="c4"
      data-testid="navbar-form-button"
      type="submit"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.023 16.977a35.13 35.13 0 0 1-1.367-1.384c-.372-.378-.596-.653-.596-.653l-2.8-1.337A6.962 6.962 0 0 0 16 9c0-3.859-3.14-7-7-7S2 5.141 2 9s3.14 7 7 7c1.763 0 3.37-.66 4.603-1.739l1.337 2.8s.275.224.653.596c.387.363.896.854 1.384 1.367l1.358 1.392.604.646 2.121-2.121-.646-.604c-.379-.372-.885-.866-1.391-1.36zM9 14c-2.757 0-5-2.243-5-5s2.243-5 5-5 5 2.243 5 5-2.243 5-5 5z"
        />
      </svg>
    </button>
  </form>
</nav>
`;
